### ISSUE TITLE
VaadinServlet does not support Content-Range

### ISSUE THREAD

**Issue by AustinShalit** [vaadin/flow#7465]
When using `StreamResource`, I expect that the VaadinServlet would support Content Range in the HTTP request parameters but it does not.

This is particularly an issue if the resource is a video file and the user is using Safari as discussed in Apple's documentation here: https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariWebContent/CreatingVideoforSafarioniPhone/CreatingVideoforSafarioniPhone.html#//apple_ref/doc/uid/TP40006514-SW6

**Comment by TatuLund** [comment]
Just for information, the same feature was missing also in Vaadin 8, thus I wrote [AudioVideo add-on](https://vaadin.com/directory/component/audiovideo) that adds support for this feature. Technically it is possible to add custom request handlers or override the handleRequest fof the VaadinService. I think it  should be possible to apply the logic there. You can use my implementation as a reference

https://github.com/TatuLund/audiovideo/blob/master/gwtav-addon/src/main/java/org/vaadin/gwtav/AbstractAudioVideo.java#L100

And

https://github.com/TatuLund/audiovideo/blob/master/gwtav-addon/src/main/java/org/vaadin/gwtav/IOUtil.java#L24

**Comment by chkpnt** [comment]
Isn't it a major bug that `StreamResource` simply doesn't work in Safari?

**Comment by TatuLund** [comment]
@mstahv , yes Tomcat and Jetty throw different exception when browser detects that range request is supported. I.e. connection is aborted and index table is fetched before restart, see here https://github.com/TatuLund/audiovideo/blob/master/gwtav-addon/src/main/java/org/vaadin/gwtav/IOUtil.java#L105

**Comment by chkpnt** [comment]
> @chkpnt Did you also want to serve videos? From a filesystem or from database?

Yes, it's about videos from the filesystem. Luckily, our clients doesn't use Safari.
